src/blockdiagonal.jl

```diff
@@ -119,7 +119,20 @@ end
 
 Base.collect(B::BlockDiagonal) = Matrix(B)
 
-Base.size(B::BlockDiagonal) = sum(first∘size, blocks(B)), sum(last∘size, blocks(B))
+# init kwarg for sum introduced in v1.6
+# see https://github.com/JuliaLang/julia/blob/master/base/reduce.jl#L492
+@static if VERSION  < v"1.6"
+    function Base.size(B::BlockDiagonal)
+        if isempty(blocks(B))
+            return 0, 0
+        else
+            return sum(first∘size, blocks(B)), sum(last∘size, blocks(B))
+        end
+    end
+else
+   Base.size(B::BlockDiagonal) = sum(first∘size, blocks(B); init=0), sum(last∘size, blocks(B); init=0)
```

Comment:
We should add tests for kwargs version.

Btw, how come the new version does not need to check whether it is empty? e.g. I get in Julia 1.8
```julia
julia> sum(first∘size, [])
ERROR: MethodError: reducing over an empty collection is not allowed; consider supplying `init` to the reducer
```

Reply:
The kwargs version should be tested by the Julia 1 CI (which currently is 1.8), and non kwargs by 1.0 CI?

Setting `init=0` will mean `sum` will return `0` when reducing over an empty collection, so in Julia 1.8 

```Julia
julia> sum([];init=0)
0
```

Reply:
Ah, sorry, I _somehow_ misread as the `init` kwarg was added to the `size` 😅 i don't know how really...